### ISSUE TITLE
508-defect-2 [MOBILE-DESIGN]: Mobile layout SHOULD be legible and easy-to-use

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#12422]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

On the mobile device tested (iPhone 8+), the table layout would benefit from adopting the new responsive table layout (started by the VSA eBenefits team). The primary concern is that `Download` and `letter` run together in the 

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As an iOS user, I want to understand the text of the links to download the various debt letter PDFs.

## Environment

* Operating System: iOS (tester used an iPhone 8+)
* Browser: Safari
* Server destination: staging

## Steps to Recreate

1. Log in as Greg Anderson (user 1)
1. Enter https://staging.va.gov/manage-va-debt/your-debt/ in your browser
1. Verify that the table layout could be improved for readability
1. Verify that `Download` and `letter` run together
[see screen shot below]
Please note: the behavior with zoom on desktop does not replicate this issue

## Possible Fixes (optional)

1. Consider using the new responsive table component (I asked in DSVA Slack design channel to find out the status of the design system component, but maybe it is possible to ask the eBenefits team for their code.)
1. Ensure that the words `Download` and `letter` do not merge on mobile
1. Consider the redesign suggested in https://github.com/department-of-veterans-affairs/va.gov-team/issues/12192

## WCAG or Vendor Guidance (optional)

* [W3C WCAG 2.0 1.3.1 Adaptable](https://www.w3.org/WAI/GL/mobile-a11y-tf/MobileTechniques/#content-structure-separation) - Create content that can be presented in different ways (for example simpler layout) without losing information or structure.

* [W3C WCAG 2.0 3.1 Readable](https://www.w3.org/WAI/GL/mobile-a11y-tf/MobileTechniques/#meaning) - Make text content readable and understandable.

## Screenshots or Trace Logs

![IMG_8715](https://user-images.githubusercontent.com/57469/90042025-201afa00-dc98-11ea-9967-3a2bb7c8f0f5.PNG)

**Comment by bradl3yC** [comment]
@jenstrickland I was not able to see this - but I did shift the right rail content which may have corrected this.  Can we re-validate?

**Comment by jenstrickland** [comment]
@bradl3yC I am still seeing this. Are you able to test using an iPhone? "Download" and "letter" blend together in both portrait and landscape modes.

![letter2](https://user-images.githubusercontent.com/57469/90567002-65976580-e177-11ea-959c-93ff9c4d6d9d.png)

![letter1](https://user-images.githubusercontent.com/57469/90567013-6af4b000-e177-11ea-8953-8b6cd6042ecb.png)

**Comment by bradl3yC** [comment]
@jenstrickland I tested on both an iphone and xcode simulator which simulates an iphone using safari and I did not have this experience

**Comment by leahkeeler** [comment]
@jenstrickland - Can you please verify? advise on next steps or close ticket -Thanks, Leah

**Comment by jenstrickland** [comment]
@leahkeeler I just logged in as user 1, Greg, a couple of times, and am seeing:

1. "Our records show that you don't have any current debts", so the table doesn't show. 
2. "Your debt letters are currently unavailable."

Is there an issue, and I should try again in a bit? 

cc @1Copenut — have you ever seen this sort of an issue on iOS, where the words in a link smoosh together?

**Comment by bradl3yC** [comment]
@jenstrickland we hid all resolved debts as per the AC - the DMC is working to upgrade their test data to include a current debt that has a balance greater than 0 dollars.

**Comment by 1Copenut** [comment]
@jenstrickland I'm guessing the issue could be caused by having `<span>` tags co-mingled with the text inside the link:

```html
<a
  href="https://staging-api.va.gov/v0/debt_letters/%7B64B0BDC4-D40C-4C54-86E0-104C987B8D8F%7D"
  rel="noopener noreferrer"
  target="_blank"
>
  <i
    aria-hidden="true"
    class="fas fa-download vads-u-padding-right--1"
    role="img">
  </i>
  Download
  <span class="sr-only">Second Demand Letter</span>
  letter
  <span class="sr-only">dated May 29, 2020</span>
  <dfn>
    <abbr title="Portable Document Format">(PDF)</abbr>
  </dfn>
</a>
```

Could the spans be rewritten in such a way that users are only reading one continuous string?

```html
<a href="/" ... >
  <i />
  <span aria-hidden="true">Download letter (PDF)</span>
  <span class="sr-only">Download Second Demand Letter dated May 29, 2020</span>
  <dfn>...</dfn>
</a>
```

**Comment by bradl3yC** [comment]
@1Copenut I can try that - give me a few and Ill have a PR up

**Comment by bradl3yC** [comment]
@jenstrickland @1Copenut this should be in staging in about 30 minutes or so

**Comment by jenstrickland** [comment]
@1Copenut — brilliant, that was it! Thanks, @bradl3yC — validated! 🎉 

![IMG_8819](https://user-images.githubusercontent.com/57469/90684839-da7da480-e236-11ea-8983-b3f385b08e73.PNG)

![IMG_8820](https://user-images.githubusercontent.com/57469/90684840-da7da480-e236-11ea-81c8-97921e9fbe0e.PNG)

**Comment by 1Copenut** [comment]
Outstanding @jenstrickland! Glad to hear that was the right update.
